services/email_verification.js

```diff
@@ -35,45 +41,57 @@ const mailer = require("nodemailer").createTransport({
 });
 
 const server = micro(async req => {
-    const data = await json(req);
+    const data = await json(req.body);
     // generate a token, add it to redis
     let id = idGen.nextId();
-    client.hset(data.user.name, `email_verification`, JSON.stringify({
-        deleteThis: true,
-        id
-    }), async err => {
+    let token = bcrypt.hashSync(`${data.user.email}:${id}`)
+    client.set(`${data.user.email}:email_verify`, token, async err => {
 
-        if (err) send(res, 500, `{code: 500, message: ${err}`);
+        if (err) send(res, 500, `{"code": "500", "message": ${err}`);
```

Comment:
Missing an end bracket on this JSON response. Maybe just do `JSON.stringify({blah blah blah})` instead of forming it yourself.